### PR TITLE
Release notes of k8s provider v2.1.0

### PR DESCRIPTION
**Pull request by linglingye001**
draft release notes for v2.1.0

### CODE REVIEW

File: releaseNotes/KubernetesProvider.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing ConfigMap without using sentinel keys. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
+* Added support for refreshing Secrets only if data from Key Vault is updated. This enhancement can reduce the overall resource overhead and improve the performance of your workload.
+
+### Bug fixes
+* Fixed a bug where empty string label filter is not converted to a null label. [#67](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/67)
```
Comment (RichardChen820) [inline releaseNotes/KubernetesProvider.md:13]: https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/67
This thread is very easy to lead to confusion, how about create another issue to explain what is the bug?

File: releaseNotes/KubernetesProvider.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 [Image][image] | [Sample][sample]
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing ConfigMap without using sentinel keys. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
```
Comment (RichardChen820) [inline releaseNotes/KubernetesProvider.md:9]: Add support for refreshing configMap by watching all key-values. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. Watching sentinel key to refresh is still supported, but it's not required now.

File: releaseNotes/KubernetesProvider.md [hunk 3]
```diff
@@ -2,6 +2,16 @@
 
 [Image][image] | [Sample][sample]
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing ConfigMap without using sentinel keys. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
+* Added support for refreshing Secrets only if data from Key Vault is updated. This enhancement can reduce the overall resource overhead and improve the performance of your workload.
```
Comment (RichardChen820) [inline releaseNotes/KubernetesProvider.md:10]: > Added support for refreshing Secrets only if data from Key Vault is updated

> Added support for refreshing ConfigmMap and Secrets only if their data is really updated.

File: releaseNotes/KubernetesProvider.md [hunk 4]
```diff
@@ -2,6 +2,16 @@
 
 [Image][image] | [Sample][sample]
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing configMap by watching all key-values. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. Watching sentinel key to refresh is still supported, it's optional now. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
+* Added support for refreshing ConfigMaps and Secrets only if their data is really updated.
```
Comment (zhenlan) [inline releaseNotes/KubernetesProvider.md:10]: Let's move this to bug fix:

Fix a bug where ConfigMaps and Secrets may be regenerated even if there is no data change.

File: releaseNotes/KubernetesProvider.md [hunk 5]
```diff
@@ -2,6 +2,16 @@
...
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing configMap by watching all key-values. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. Watching sentinel key to refresh is still supported, it's optional now. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
+* Added support for refreshing ConfigMaps and Secrets only if their data is really updated.
+
+### Bug fixes
+* Fixed a bug where empty string label filter is not converted to a null label. [#76](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/76)
```
Comment (zhenlan) [inline releaseNotes/KubernetesProvider.md:13]: Fixed a bug where key-values with no label are not loaded if the label filter is specified as empty string.

File: releaseNotes/KubernetesProvider.md [hunk 6]
```diff
@@ -2,6 +2,16 @@
 
 [Image][image] | [Sample][sample]
 
+## 2.1.0 - 
+
+### Enhancements
+* Added support for load balancing mode, which enables your workloads to distribute requests to App Configuration across all available replicas. This enhancement improves the scalability of applications that typically experience high request volumes to App Configuration, ensuring they remain within quota limits. Load balancing mode is disabled by default and can be activated by setting `loadBalancingEnabled` to `true`. [#55](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/55)
+* Added support for refreshing configMap by watching all key-values. The Kubernetes provider will monitor the changes of selected key-values/feature flags and refresh the target ConfigMap only if the data updates. Watching sentinel key to refresh is still supported, it's optional now. [#40](https://github.com/Azure/AppConfiguration-KubernetesProvider/issues/40)
```
Comment (zhenlan) [inline releaseNotes/KubernetesProvider.md:9]: Added support for monitoring all selected key-values. The target ConfigMaps and Secrets will be refreshed if any of key-values are updated. Watching the sentinel key for refresh helps ensure data integrity of configuration changes but it's now optional.

**Review by zhenlan — DISMISSED** [review]

**Review by RichardChen820 — APPROVED** [review]

**Pull request merged**